### ISSUE TITLE
Add SSL Verification

### ISSUE THREAD

**Issue by gbischoff-flexcon** [airvzxf/ftp-deployment-action#8]
Current version disables SSL verification, could this be defaulted on and add it is a parameter for those that want to enable insecure connections

**Comment by airvzxf** [comment]
Thank you for your suggestion. I'll work on this suggestion during this weekend.

**Comment by airvzxf** [comment]
Hi, @gbischoff-flexcon. Regarding the official documentation (https://lftp.yar.ru/lftp-man.html). Would you kindly help to specify what setting are you talking about?

```
lftp  can  handle  several file access methods - FTP, FTPS, HTTP, HTTPS, HFTP, FISH, SFTP and
file (HTTPS and FTPS are only available when  lftp  is  compiled  with  GNU  TLS  or  OpenSSL
library).
```

```
If lftp was compiled with OpenSSL (configure --with-openssl) it includes  software  developed
by the OpenSSL Project for use in the OpenSSL Toolkit. (http://www.openssl.org/)

```

```
ftp:ssl-allow (boolean)
      if  true,  try  to  negotiate SSL connection with FTP server for non-anonymous access.
      Default is true. This and other SSL settings are only available if lftp  was  compiled
      with an ssl/tls library.

ftp:ssl-auth (string)
      the  argument for AUTH command, can be one of SSL, TLS, TLS-P, TLS-C.
      See [RFC4217](https://tools.ietf.org/html/rfc4217)) for
      explanations. By default TLS or SSL will be used, depending on FEAT reply.

ftp:ssl-data-use-keys (boolean)
      if true, lftp loads ssl:key-file for protected data connection  too.  When  false,  it
      does  not,  and  the  server  can  match  data  and control connections by session ID.
      Default is true.

ftp:ssl-force (boolean)
      if true, refuse to send password in clear when server does not support  SSL.   Default
      is false.

ftp:ssl-protect-data (boolean)
      if  true,  request SSL connection for data transfers. This provides privacy and transвЂђ
      mission error correction. Was cpu-intensive on old CPUs. Default is true.

ftp:ssl-protect-fxp (boolean)
      if true, request SSL connection for data transfer between two FTP servers in FXP mode.
      CPSV  or SSCN command will be used in that case. If SSL connection fails for some reaвЂђ
      son, lftp would try unprotected FXP transfer unless ftp:ssl-force is set  for  any  of
      the two servers. Default is true.

ftp:ssl-protect-list (boolean)
      if true, request SSL connection for file list transfers. Default is true.

ftp:ssl-use-ccc (boolean)
      if  true,  lftp would issue CCC command after logon, thus disable ssl protection layer
      on control connection.
```

```
ssl:ca-file (path to file)
      use specified file as Certificate Authority certificate.

ssl:ca-path (path to directory)
      use specified directory  as  Certificate  Authority  certificate  repository  (OpenSSL
      only).

ssl:check-hostname (boolean)
      when  true,  lftp checks if the host name used to connect to the server corresponds to
      the host name in its certificate.

ssl:crl-file (path to file)
      use specified file as Certificate Revocation List certificate.

ssl:crl-path (path to directory)
      use specified directory as Certificate Revocation List certificate repository (OpenSSL
      only).

ssl:key-file (path to file)
      use  specified  file as your private key. This setting is only used for ftps and https
      protocols. For sftp and fish protocols use sftp:connect-program and  fish:connect-proвЂђ
      gram respectively (add `-i' option to ssh).

ssl:cert-file (path to file)
      use specified file as your certificate.

ssl:use-sni (boolean)
      when true, use Server Name Indication (SNI) TLS extension.

ssl:verify-certificate (boolean)
      if  set  to  yes, then verify server's certificate to be signed by a known Certificate
      Authority and not be on Certificate Revocation List. You can specify either host  name
      or certificate fingerprint in the closure.

ssl:priority (string)
      free  form priority string for GnuTLS. If built with OpenSSL the understood values are
      + or - followed by SSL3.0, TLS1.0, TLS1.1 or TLS1.2, separated by :. Example:
      set ssl:priority "NORMAL:-SSL3.0:-TLS1.0:-TLS1.1:+TLS1.2"
```

**Comment by airvzxf** [comment]
I'm refactoring, and I'm adding a new parameter called `lftp_settings`, which is an empty string where anyone can add whatever settings they want to add. For example: `lftp_settings: "set ssl:verify-certificate true; set check-hostname yes; set ssl:ssl-force 1; set ssl:key-file '/home/wolf/.ssh/rovisoft.key'"`

In this case, you can do the following:

```yaml
name: CI -> Deploy to My website
on:
...
      - name: Upload from public_html via FTP
        uses: airvzxf/ftp-deployment-action@latest
        with:
          server: ${{ secrets.FTP_SERVER }}
          user: ${{ secrets.FTP_USERNAME }}
          password: ${{ secrets.FTP_PASSWORD }}
          local_dir: "./public_html"
          remote_dir: "/www/sub-domain/games/myself"
          delete: "true"
          max_retries: "7"
          no_symlinks: "false"
          lftp_settings: "set ssl:verify-certificate true; set check-hostname yes; set ssl:ssl-force 1; set ssl:key-file '/home/wolf/.ssh/rovisoft.key'"
```

I think that it is the best approach, rather than creating every setting parameter in my application (wrapper). If I release this version, do you think it would be helpful and useful to you?

**Comment by airvzxf** [comment]
Last changes were released in the version 1.3.0. Would you mind confirming if it works for you?

https://github.com/airvzxf/ftp-deployment-action/releases/tag/v1.3.0

```yaml
- name: FTP Deployment
  uses: airvzxf/ftp-deployment-action@v1.3.0
```

**Comment by airvzxf** [comment]
I'll close this issue. If you encounter any issue with this new version or the `lftp_settings` parameter, please create a new issue and I check it further.
